phoenix-core/src/main/java/org/apache/phoenix/schema/KeyValueSchema.java

```diff
@@ -121,7 +122,15 @@ public boolean isNull(int position, ValueBitSet bitSet) {
                         int nBytes = ptr.getLength();
                         b = ensureSize(b, offset, offset + nBytes);
                         System.arraycopy(ptr.get(), ptr.getOffset(), b, offset, nBytes);
-                        offset += nBytes;
+                        if (field.getMaxLength() != null && field.getMaxLength() > 0 ) {
```

Comment:
Is this change required to handle existing data of type char after we increase the length? We should only do this padding if the type is PChar right? Is it possible to change the length of any other fixed length data type?

Reply:
@twdsilva  Yes, we should only pad Pchar type, through other data types of fixed width which excepts for PChar have a permanent length of columns. But in this cases that cast type of fixed columns, the results may be incorrect.